zmon_worker_extras/check_plugins/openam.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import requests
+import os
+
+from zmon_worker_monitor.zmon_worker.errors import CheckError
+from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+
+class OpenAMFactory(IFunctionFactoryPlugin):
+    def __init__(self):
+        super(OpenAMFactory, self).__init__()
+
+    def configure(self, conf):
+        """
+        Called after plugin is loaded to pass the [configuration] section in their plugin info file
+        :param conf: configuration dictionary
+        """
+        self.openam_base_url = config.get('url')
+        return
+
+    def create(self, factory_ctx):
+        """
+        Automatically called to create the check function's object
+        :param factory_ctx: (dict) names available for Function instantiation
+        :return: an object that implements a check function
+        """
+        return propartial(OpenAMWrapper, url=self.openam_base_url)
+
+class OpenAMError(CheckError):
+    def __init__(self, message):
+        self.message = message
+        super(OpenAMError, self).__init__()
+
+    def __str__(self):
+        return 'OpenAM Error. Message: {}'.format(self.message)
+
+
+class OpenAMWrapper(object):
+    def __init__(
+            self,
+            url,
+            params=None,
+            timeout=10,
+            max_retries=0,
+            verify=True,
+            headers=None,
+            ):
+        self.url = url
+        self.params = params or {}
+        self.headers = headers or {}
+
+    def auth(self, chain=None, realm=None, user=os.getenv("OPENAM_USER"), password=os.getenv("OPENAM_PASSWORD")):
```

Comment:
Isn't it possible to set ``OPENAM_USER`` and ``OPENAM_PASSWORD`` in plugin config similar to ``url``? Just to be consistent with the plugin system.